### ISSUE TITLE
ingame commands don't work on 1.19.1

### ISSUE THREAD

**Issue by powercasgamer** [LuckPerms/LuckPerms#3440]
# **Temporarily Fix**

Set `register-command-list-data` to `false` in `config.yml`

### Description

Commands in game don't work as of Paper 1.19.1. Commands in console do still work. 

### Reproduction Steps

Download Paper 1.19.1 (build 86)
Download LuckPerms 5.4.36

Type a command ingame i.e `/lp info`
Nothing logged to console, ingame you get a error with a NoOpperationException hover

![2022-07-28_18 12 52](https://user-images.githubusercontent.com/40437205/181590938-b7dda1f0-1890-4ed1-a44d-dc962e01edda.png)

### Expected Behaviour

For it to work?

### Server Details

Paper version git-Paper-86 (MC: 1.19.1) (Implementing API version 1.19.1-R0.1-SNAPSHOT) (Git: 4a4ee79)

### LuckPerms Version

v5.4.36

### Logs and Configs

https://pastes.dev/u0SqevbKMW

https://pastes.dev/BI7Pxb6HEN (w/ debug enabled in server.properties and spigot.yml)

### Extra Details

Setting `register-command-list-data` to `false` "fixes" it

I assume it's related to https://github.com/lucko/commodore/issues/27

**Comment by mrfloris** [comment]
I can reproduce this, on the .36 build, and I don't have anything in console, it only says this in-game (the red msg). So I am not sure what the full error is :)

**Comment by jar349** [comment]
I will note that running `/lp` commands in the console _does_ work for me.

**Comment by dinobossytnew** [comment]
yes bug.

**Comment by Tazler89** [comment]
Same thing here nothing /lp works in chat only console but my error is not the same
![image](https://user-images.githubusercontent.com/51469267/181812781-8969e97b-c4cf-4b88-b264-19e6001b2a96.png)

**Comment by powercasgamer** [comment]
> Same thing here nothing /lp works in chat only console but my error is not the same
> 
> ![image](https://user-images.githubusercontent.com/51469267/181812781-8969e97b-c4cf-4b88-b264-19e6001b2a96.png)
> 
> 

Set `register-command-list-data` to false in config.yml, also try updating your paper version

**Comment by jar349** [comment]
> Set register-command-list-data to false in config.yml

I confirm that this makes `/lp` commands in-game work for me again

**Comment by andrewkm** [comment]
Same thing here.
No console error.
Console commands work fine, in-game commands do not.

Paper 1.19.1 - Build 91
Luckperms - Version 5.4.36

**Comment by Tazler89** [comment]
> > Same thing here nothing /lp works in chat only console but my error is not the same
> > ![image](https://user-images.githubusercontent.com/51469267/181812781-8969e97b-c4cf-4b88-b264-19e6001b2a96.png)
> 
> Set `register-command-list-data` to false in config.yml, also try updating your paper version

paper was fully up to date, but thank you for now that seems to have fixed it, you must restart after this change not just reload will not work, Thank you, Fix Confirmed

**Comment by Gu-f** [comment]
> 这里也一样。 没有控制台错误。 控制台命令工作正常，游戏内命令不行。
> 
> 论文 1.19.1 - Build 91 Luckperms - 版本 5.4.36



> Same thing here. No console error. Console commands work fine, in-game commands do not.
> 
> Paper 1.19.1 - Build 91 Luckperms - Version 5.4.36

I have the same problem

**Comment by Mikkel136** [comment]
Can confirm on Purpur build 1747
Commands specifically does not work through player chat event.

**Comment by sayak-brm** [comment]
Can reproduce:

```
[18:22:09 INFO]: [LP] Running LuckPerms v5.4.36 by Luck.
[18:22:09 INFO]: [LP] -  Platform: Bukkit
[18:22:09 INFO]: [LP] -  Server Brand: Paper
[18:22:09 INFO]: [LP] -  Server Version:
[18:22:09 INFO]: [LP]      git-Paper-97 (MC: 1.19.1) - 1.19.1-R0.1-SNAPSHOT
[18:22:09 INFO]: [LP] -  Storage:
[18:22:09 INFO]: [LP]      Type: MySQL
[18:22:09 INFO]: [LP]      Ping: 76ms
[18:22:09 INFO]: [LP]      Connected: true
[18:22:09 INFO]: [LP] -  Extensions:
[18:22:09 INFO]: [LP]      None
[18:22:09 INFO]: [LP] -  Messaging: PluginMessage
[18:22:09 INFO]: [LP] -  Instance:
[18:22:09 INFO]: [LP]      Static contexts: None
[18:22:09 INFO]: [LP]      Online Players: 0 (0 unique)
[18:22:09 INFO]: [LP]      Uptime: 2m 31s
[18:22:09 INFO]: [LP]      Local Data: 0 users, 5 groups, 0 tracks```

Note that `/lpb` works on BungeeCord and the issue is only for the `/lp` command.

**Comment by lucko** [comment]
Hey everyone, thanks for your patience with this one. I wasn't able to get to my computer for the past few days - apologies once again for the delay.

The somewhat-hacky code (oops 😄) that caused this issue has now been replaced with an alternative implementation that uses a proper API provided by Paper-derived bukkit servers. (see https://github.com/lucko/commodore/commit/98f39247bf37ae5e754589a60bfd0d6f5c35533d for details) Shouldn't happen again!

The fix is available in v5.4.37 and newer :)
